### PR TITLE
[Northumberland] Fetch assigned-to users

### PR DESCRIPTION
**Pull request by nephila-nacrea**
Note: also need to deploy changes to northumberland alloy config on mysociety-servers.

This is the open311-adapter portion of https://github.com/mysociety/societyworks/issues/3978. We look for assigned-to users in each update from Alloy and pass email and name through to FMS via the 'extras' field.

[skip changelog]

### CODE REVIEW

File: perllib/Open311/Endpoint/Integration/AlloyV2.pm [hunk 1]
```diff
@@ -633,13 +636,39 @@ sub _get_inspection_updates {
                 updated_datetime => $update_dt,
             );
 
+            my $assigned_to_user = {};
```
Comment (dracos) [inline perllib/Open311/Endpoint/Integration/AlloyV2.pm:639]: Don't think this line is needed, could be `my $assigned_to_user` inside the if.

File: perllib/Open311/Endpoint/Integration/AlloyV2.pm [hunk 2]
```diff
@@ -633,13 +636,39 @@ sub _get_inspection_updates {
...
+            if ( my $assigned_to_user_id
+                = $attributes->{ $mapping->{assigned_to_user} // '' }[0] )
+            {
+                $assigned_to_user
```
Comment (dracos) [inline perllib/Open311/Endpoint/Integration/AlloyV2.pm:643]: This block is fine, wondered if you'd like to rejig it to something like:

```
my $assigned_to_user = $assigned_to_users->{$assigned_to_user_id} ||= do {
    # There is a possibility the assigned-to user is not already
    # in the $assigned_to_users hash; do another lookup if so
    $self->get_assigned_to_users($resource)->{$assigned_to_user_id};
};
```
which then handles the checking/reassignment in a neater block.

**Review by dracos — CHANGES_REQUESTED** [review]
Looks good, all clear to read 👍  Couple of small suggestions to the block is all :)

File: perllib/Open311/Endpoint/Integration/AlloyV2.pm (hunk 2 above)
Comment (nephila-nacrea) [reply perllib/Open311/Endpoint/Integration/AlloyV2.pm:643]: Ooh nice and pithy, I wasn't that familiar with using do{} in this kind of way :-)

**Review by dracos — APPROVED** [review]
🧑‍🚀

**Pull request merged**